### PR TITLE
use vector of vectors instead of unique pointer

### PR DESCRIPTION
**Pull request by sina-rostami**
First I used one vector instead of unique pointer, I faced with 
```
cannot convert ‘std::vector<bool>::reference*’ {aka ‘std::_Bit_reference*’} to ‘bool*’ in initialization
```
in index operator overloading.
then I decided to use vector of vectors and I think it's more readable in code.
but a bit slower...

### CODE REVIEW

File: Array2D.h [hunk 1]
```diff
@@ -1,37 +1,36 @@
...
-using std::unique_ptr;
+#include <vector>
+using std::size_t;
+using std::vector;
 
 template <typename T> class Array2D {
 private:
   std::size_t height, width;
-  unique_ptr<T[]> arr_ptr;
+  vector<vector<T>> vc;
```
Comment (IYP-Programer-Yeah) [inline Array2D.h:10]: No, consider vector<T> instead of vector<vector<t>>.
Contiguous memory.

**Review by IYP-Programer-Yeah — CHANGES_REQUESTED** [review]

File: Array2D.hpp [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+  bool operator==(const Array2D<T> &other) const {
+    if (width != other.width || height != other.height)
+      return false;
+    return vc == other.vc;
+  }
+
+  bool operator!=(const Array2D<T> &other) const { return !operator==(other); }
+
+  Array2D<T> &operator=(const Array2D<T> &other) {
```
Comment (IYP-Programer-Yeah) [inline Array2D.hpp:37]: not needed

File: Array2D.hpp [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  std::size_t height, width;
+  vector<T> vc;
+
+public:
+  Array2D() : Array2D(0, 0){};
+
+  Array2D(const size_t height1, const size_t width1) // constructor
+      : height(height1), width(width1), vc(vector<T>(height * width)) {}
+
+  Array2D(const Array2D<T> &other)
```
Comment (IYP-Programer-Yeah) [inline Array2D.hpp:20]: not needed

File: Gol.cpp [hunk 4]
```diff
@@ -1,8 +1,11 @@
-#include "World.h"
+#include "Universe.hpp"
+#include <unistd.h>
```
Comment (IYP-Programer-Yeah) [inline Gol.cpp:2]: what is this included for?

File: Gol.cpp [hunk 5]
```diff
@@ -25,16 +28,18 @@ class GameOfLife {
   }
 
   Result solve() {
-    std::size_t max_size = 1 << first_world.get_height() * first_world.get_width();
+    std::size_t max_size =
```
Comment (IYP-Programer-Yeah) [inline Gol.cpp:31]: const?

File: Array2D.hpp [hunk 6]
```diff
@@ -0,0 +1,64 @@
+#include <iostream>
+#include <iterator>
+#include <memory>
+#include <vector>
+
+using std::size_t;
+using std::vector;
```
Comment (IYP-Programer-Yeah) [inline Array2D.hpp:7]: almost always avoid using, they are rarely OK.

**Review by IYP-Programer-Yeah — CHANGES_REQUESTED** [review]

File: Gol.cpp (hunk 4 above)
Comment (sina-rostami) [reply Gol.cpp:2]: I was testing something

File: Array2D.hpp (hunk 3 above)
Comment (sina-rostami) [reply Array2D.hpp:20]: which one? copy constructor?

File: Gol.cpp (hunk 5 above)
Comment (sina-rostami) [reply Gol.cpp:31]: yes it's better to be const...

File: Array2D.hpp (hunk 6 above)
Comment (sina-rostami) [reply Array2D.hpp:7]: OK

File: Array2D.hpp [hunk 7]
```diff
@@ -1,62 +1,38 @@
...
 
   bool operator==(const Array2D<T> &other) const {
-    if (width != other.width || height != other.height)
-      return false;
-    return vc == other.vc;
-  }
-
-  bool operator!=(const Array2D<T> &other) const { return !operator==(other); }
-
-  Array2D<T> &operator=(const Array2D<T> &other) {
```
Comment (sina-rostami) [inline Array2D.hpp]: I think it is ok to use compiler generated copy assignment for the case.

File: .vscode/settings.json [hunk 8]
```diff
@@ -0,0 +1,5 @@
+{
+    "files.associations": {
```
Comment (IYP-Programer-Yeah) [inline .vscode/settings.json:2]: You probably dont wanna push this

File: Gol.cpp [hunk 9]
```diff
@@ -1,70 +1,71 @@
...
-  World do_evolutions(World temp_world) {
+  Universe::World do_evolutions(Universe::World temp_world) {
     for (std::size_t i = 0; i < evolution_num; i++)
       temp_world.evolution();
     return temp_world;
   }
 
   Result solve() {
-    std::size_t max_size = 1 << first_world.get_height() * first_world.get_width();
+    const std::size_t MAX_SIZE =
```
Comment (IYP-Programer-Yeah) [inline Gol.cpp:26]: Why did you change the naming convention?

**Review by IYP-Programer-Yeah — CHANGES_REQUESTED** [review]

**Pull request merged**